README.md

```diff
@@ -0,0 +1,219 @@
+
+They are fundamental to the operation of nearly every contemporary website.
+
+Primarily, cookies are used for log in. They provide a way for us to verify who a user is once, and then remember it for their entire session. Without cookies, you would have to provide your username and password on every single request to the server.
+
+Cookies may also be used to store other information about a user, such as what's in their shopping cart, or what ads you've shown them during their visit.
+
+In this document, we'll cover what cookies are, how they fit into the HTTP response flow, and how you can access them from Rails.
+
+## Objectives
+  1. Describe a cookie.
+  2. Explain the data flow problem cookies solve.
+  3. Find all the cookies on a page.
+  4. Explain what a session is, as far as Rails is concerned.
+  5. Describe where session data is stored.
+  6. Write data to the Rails session and access it later
+     in the app.
+
+## Shopping
+
+Let's say we want to write an e-commerce site. Users can come to the site, add some items to their shopping cart, and then pay for all of them at the end. Simple.
+
+Where does the user's shopping cart live?
+
+We could create a `Cart` model, which `has_and_belongs_to_many` `Item`s. But that leaves a fundamental problem: how do we know what cart to load when we get a request?
+
+Remember what's included in an HTTP request:
+  * an [HTTP verb][rfc_http_methods], like `GET`, `PUT`, or `POST`
+  * a path
+  * various headers
+
+HTTP servers are typically stateless. Your Rails server has access to a database, but the request needs to have information about what database rows to look up.
+
+For example, `GET` requests usually encode this information in the path. When you write a route matching `/items/:item_id`, you are telling Rails to pull the value `id` from the request path and save it in `params[:id]`. In your `items_controller`, you'll probably have a method that looks something like:
+
+    def show
+      @item = Item.find!(params[:item_id])
+    end
+
+Which loads the row for that item from the database and returns it as an ActiveRecord model object, which your `show.html.erb` then renders.
+
+If we want to be able to retrieve the current cart, we need to its id to be somewhere in the HTTP request. Specifically, it must be in the path or the headers.
+
+It would be possible, though quite convoluted, to store this information in the path. This would have strange effects: since the path is shown in the browser's URL bar, a user who copies a URL and sends it to a friend ("check out this neat skirt!") would also be copying their shopping cart ID. Upon loading the page, the friend would
+see what's in the user's cart. That is probably not what we want.
+
+Cookies allow us to store this information in the only other place available to us: HTTP headers.
+
+## What's a cookie, anyway?
+
+Let's see what [the spec][rfc_cookies] has to say:
```

Comment:
We generally shy away from the super technical speak of specs and such. I think it's fine to keep this in here if you'd like, but then after this I'd mention that we don't expect beginners to be able to understand everything. Then go into explaining at how it actually works.